pkgs/games/natural-pong/default.nix

```
@@ -0,0 +1,41 @@
+{ lib, udev, alsa-lib, vulkan-loader, xorg, libxkbcommon, wayland, pkg-config, makeWrapper, rustPlatform, fetchFromGitHub }:
+let
+  buildInputs = [
+    udev alsa-lib vulkan-loader
+    xorg.libX11 xorg.libXcursor xorg.libXrandr xorg.libXi
+    libxkbcommon wayland
+  ];
+  nativeBuildInputs = [
+    pkg-config
+    makeWrapper
+  ];
+in rustPlatform.buildRustPackage {
+  pname = "natural-pong";
+  version = "0.1.0";
+  cargoHash = "sha256-jfeQjZrGyGXQ298naakiDpSlbl116ZKyi0RBkLnlwsA=";
+
+  buildInputs = buildInputs;
+  nativeBuildInputs = nativeBuildInputs;
+  LD_LIBRARY_PATH = lib.makeLibraryPath buildInputs;
```

I don't know rust packaging well enough to know what LD_LIBRARY_PATH does but if you really need this you can do something like:
```suggestion
  LD_LIBRARY_PATH = lib.makeLibraryPath finalAttrs.buildInputs;
```

---

`finalAttrs` was not implemented yet outside mkDerivation

---

LD_LIBRARY_PATH defines the path where the binary should search for the libraries. The application will search for several libraries like xorg and wayland.

I've used this documentation and modified it: https://github.com/bevyengine/bevy/blob/main/docs/linux_dependencies.md#nix

I guess a let statement is still required since finalAttrs doesn't work here.

---

> I guess a let statement is still required since finalAttrs doesn't work here.
you can write
```nix
rustPlatform.buildRustPackage rec {
```
this makes it recursive and you can use attributes defined in the same scope.